### PR TITLE
Optimize the tabKey handling in the Overlay (T677337)

### PR DESCRIPTION
**Pull request by DokaRus**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets/overlay.tests.js [hunk 1]
```diff
@@ -2945,6 +2946,20 @@ QUnit.test("elements under top overlay with shader have not to get focus by tab"
...
+QUnit.test("tabbable selectors should check only bounds", function(assert) {
+    var tabbableSpy = sinon.spy(selectors, "tabbable");
+    var overlay = new Overlay($("<div>").appendTo("#qunit-fixture"), {
+        visible: true,
+        shading: true,
+        contentTemplate: $("#focusableTemplate")
+    });
+    var elementsCount = $(overlay.content()).parent().find("*").length;
+
+    overlay._findTabbableBounds();
```
Comment (mrjohn0011) [inline testing/tests/DevExpress.ui.widgets/overlay.tests.js]: Don't use the private method. Use keyboard mock, press tab and check the count of tabbable method calls

**Review by mrjohn0011 — CHANGES_REQUESTED** [review]
Don't use private method in the test

**Review by mrjohn0011 — APPROVED** [review]

File: js/ui/overlay.js [hunk 2]
```diff
@@ -837,25 +837,42 @@ var Overlay = Widget.inherit({
         }
     },
 
-    _findTabbableElements: function() {
-        return this._$wrapper
-            .find("*").filter(selectors.tabbable);
+    _findTabbableBounds: function() {
+        var $elements = this._$wrapper.find("*");
+        var elementsCount = $elements.length - 1;
```
Comment (San4es) [inline js/ui/overlay.js:842]:
```
var elementsCount = $elements.length;

...

for(var i = 0; i < elementsCount; i++) {
```

Comment (San4es) [reply js/ui/overlay.js:842]: or

```
var lastElementIndex = $elements.length - 1;

...

for(var i = 0; i <= lastElementIndex; i++) {
```

**Pull request merged**